core/src/plan/join.rs

```diff
@@ -24,34 +24,30 @@ pub fn plan(schema_map: &HashMap<String, Schema>, statement: Statement) -> State
     }
 }
 
-struct Planner<'a> {
+struct JoinPlanner<'a> {
     schema_map: &'a HashMap<String, Schema>,
 }
 
-impl<'a> Planner<'a> {
-    fn query(&self, outer_context: Option<Rc<Context<'a>>>, query: Query) -> Query {
-        let Query {
-            body,
-            limit,
-            offset,
-        } = query;
-
-        let body = match body {
+impl<'a> Planner<'a> for JoinPlanner<'a> {
+    fn query(&self, outer_context: Option<Rc<Context<'a>>>, mut query: Query) -> Query {
+        query.body = match query.body {
             SetExpr::Select(select) => {
                 let select = self.select(outer_context, *select);
 
                 SetExpr::Select(Box::new(select))
             }
-            SetExpr::Values(_) => body,
+            SetExpr::Values(_) => query.body,
         };
 
-        Query {
-            body,
-            limit,
-            offset,
-        }
+        query
```

Comment:
```suggestion
    fn query(&self, outer_context: Option<Rc<Context<'a>>>, query: Query) -> Query {
        let body = match query.body {
            SetExpr::Select(select) => {
                let select = self.select(outer_context, *select);

                SetExpr::Select(Box::new(select))
            }
            SetExpr::Values(_) => query.body,
        };

        Query { body, ..query }
    }
```

Reply:
Wow, I didn't know this kind of assignment working. thanks a lot!